spring-integration-core/src/main/java/org/springframework/integration/gateway/MessagingGatewaySupport.java

```diff
@@ -543,7 +543,14 @@ private Object doSendAndReceive(Object object, boolean shouldConvert) {
 					reply instanceof ErrorMessage
 							? ((ErrorMessage) reply).getPayload()
 							: (Throwable) reply;
-			return handleSendAndReceiveError(object, requestMessage, error, shouldConvert);
+			try {
+				return handleSendAndReceiveError(object, requestMessage, error, shouldConvert);
+			}
+			catch (MessagingException me) {
+				if (me.getCause() instanceof Error) { // NOSONAR
+					throw (Error) me.getCause();
+				}
```

Comment:
Needs `throw me;` to retain existing behavior of throwing error flow exceptions to caller.

It's not clear why this try/catch is here at all; this is an `Error` on the error flow, not the original flow.